pkg/controllers/user/approuter/approuter.go

```diff
@@ -0,0 +1,21 @@
+package approuter
+
+import (
+	"context"
+	"github.com/rancher/types/config"
+)
+
+func Register(ctx context.Context, cluster *config.UserContext) {
+	secrets := cluster.Management.Core.Secrets("")
+	secretLister := cluster.Management.Core.Secrets("").Controller().Lister()
+	workload := cluster.UserOnlyContext()
+	c := &Controller{
+		ingressInterface:       workload.Extensions.Ingresses(""),
+		ingressLister:          workload.Extensions.Ingresses("").Controller().Lister(),
+		dnsClient:              NewClient(secrets, secretLister, workload.ClusterName),
+		clusterName:            workload.ClusterName,
+		managementSecretLister: secretLister,
+	}
+	workload.Extensions.Ingresses("").AddHandler("approuterController", c.sync)
+	go c.renew(ctx)
```

Comment:
just curious: whats the purpose of renew? Does dns provider require records renewal every 24 hours?

Reply:
Our coredns use etcd as the default backend,and etcd will regularly clean up the expiration records (default 10 days).To updating the expiration time,we set the default renew interval to 24 hour.